src/pages/CourseItem/CourseItem.tsx

```diff
@@ -0,0 +1,91 @@
+import { Box, Center, Container, Flex, Loader, Title } from "@mantine/core";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useCourseItemQuery } from "src/api/courses/hooks";
+import Lesson from "src/components/Lesson/Lesson";
+import VideoPlayer from "src/components/VideoPlayer/VideoPlayer";
+
+import CourseDetails from "./CourseDetails";
+
+const CourseItem = () => {
+  const params = useParams<{ id: string }>();
+
+  const { isLoading, data: course } = useCourseItemQuery(params.id || "");
+
+  const { lessons } = course || {};
+  lessons?.sort((a, b) => a.order - b.order);
+
+  const [currentLessonId, setCurrentLessonId] = useState<string>("");
+
+  const currentLesson = lessons?.find(
+    (lesson) => lesson.id === currentLessonId
+  );
+
+  useEffect(() => {
+    if (lessons && !currentLessonId) {
+      setCurrentLessonId(lessons?.[0].id || "");
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Better to set it **.eslintrc.json** 
Cause, this rule isn't working well and the hints from exhaustive-deps are usually incorrect.